linkedlist/linkedList.go

```diff
@@ -27,6 +37,40 @@ func (l *LinkedList) Find(n int) (index uint, found bool) {
 	return 0, false
 }
 
+func (l *LinkedList) Remove(index uint) bool {
+	if index >= l.length {
+		return false
+	}
+
+	if index == 0 {
+		l.head = l.head.Next
+		l.length--
+		return true
```

Comment:
if you add data in index 0 and then delete all of them at index 0
there should be nothing in the list.
but it won't delete the cached nodes and if you search with the concurrent finds it will return data as they are cached and never get deleted.